builtin/principals/oidcropc/Principals.go

```diff
@@ -0,0 +1,102 @@
+package oidcropc
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/signmykeyio/signmykey/builtin/principals/common"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+// Principals struct represents OIDC options for getting principals list from OIDC.
+type Principals struct {
+	OIDCUserinfoEndpoint string
+	OIDCUserGroupsEntry  string
+	TransformCase        string
+}
+
+type oidcUserinfo struct {
+	Oidcgroups []string `json:"oidcgroups"`
+}
+
+// Init method is used to ingest config of Principals
+func (p *Principals) Init(config *viper.Viper) error {
+	neededEntries := []string{
+		"oidcUserinfoEndpoint",
+		"oidcUserGroupsEntry",
+	}
+
+	var missingEntriesLst []string
+	for _, entry := range neededEntries {
+		if !config.IsSet(entry) {
+			missingEntriesLst = append(missingEntriesLst, entry)
+			continue
+		}
+	}
+	if len(missingEntriesLst) > 0 {
+		missingEntries := strings.Join(missingEntriesLst, ", ")
+		return fmt.Errorf("Missing config entries (%s) for Principals", missingEntries)
+	}
+
+	config.SetDefault("transformCase", "none")
+	tc := config.GetString("transformCase")
+	if tc != "none" && tc != "lower" && tc != "upper" {
+		return errors.New("transformCase config entry for Principals must be none, lower or upper")
+	}
+
+	p.OIDCUserinfoEndpoint = config.GetString("oidcUserinfoEndpoint")
+	p.OIDCUserGroupsEntry = config.GetString("oidcUserGroupsEntry")
+
+	p.TransformCase = tc
+
+	return nil
+}
+
+// Get method is used to get the list of principals associated to a specific user.
+func (p Principals) Get(user string) (principals []string, err error) {
+
+	reqInfo, err := http.NewRequest("GET", p.OIDCUserinfoEndpoint, nil)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
log.Fatal does an os.Exit(1) https://golang.org/src/log/log.go#L312 , here we just want to return the error :)

Reply:
Done